cmd/controller/controller.go

```diff
@@ -177,8 +170,22 @@ func (c *command) start(ctx context.Context) error {
 
 	var joinClient *token.JoinClient
 
-	if c.TokenArg != "" && c.needToJoin() {
-		joinClient, err = joinController(ctx, c.TokenArg, c.K0sVars.CertRootDir)
+	if c.needToJoin() {
+		if c.TokenArg == "" && c.TokenFile == "" {
+			return fmt.Errorf("token argument or --token-file is required to join")
+		}
```

Comment:
If we want to keep the same behavior as before it, it needs to be

```suggestion
	if (c.TokenArg != "" || c.TokenFile != "") && c.needToJoin() {
```

Reply:
But is that a wished behavior? :)